_posts/2018-09-26-artsy-frontend-history.markdown

```diff
@@ -0,0 +1,172 @@
+
+Our site was built with a framework called [Backbone][], which was really well-suited for our needs at the time.
+From their documentation:
+
+> Philosophically, Backbone is an attempt to discover the minimal set of data-structuring (models and collections)
+> and user interface (views and URLs) primitives that are generally useful when building web applications with
+> JavaScript. In an ecosystem where overarching, decides-everything-for-you frameworks are commonplace, and many
+> libraries require your site to be reorganized to suit their look, feel, and default behavior — Backbone should
+> continue to be a tool that gives you the _freedom_ to design the full experience of your web application.
+
+As an outsider to the web at that time, I can't comment too heavily on Backbone. It seems like the freedom
+(emphasis theirs) that they describe is a freedom from tangled jQuery code everywhere. I think our definition of
+freedom on the web frontend has evolved since then, but that's just my feeling.
+
+The other key component to our web frontend was [CoffeeScript][]. According to its documentation, "CoffeeScript is
+a little language that compiles into JavaScript", which was pretty important at the time. JavaScript in 2011 is
+very different from JavaScript today. The CoffeeScript docs also state that "JavaScript has always had a gorgeous
+heart", which I'm not sure I'd agree with to be honest, but the CoffeeScript project really shows how a handful of
+engineers working to improve something they care about can change and entire industry. While I don't think
+contemporary JavaScript would have gotten as good as it has without CoffeeScript, it's a bit anachronistic to see
+it used today.
+
+Our goal as a (very small!) engineering team at the time was to keep our moving parts to a minimum.
+Rails+SASS+CoffeeScript+Backbone helped us achieve that goal, and we couldn't have gotten this far without the help
+of those projects.
+
+## Ezel
+
+In November 2013, we split our web frontend from the API backend. You can read
+[all the details in this blog post](2013_review), but the story is summarized nicely as "moving from a single
+monolithic application to modular Backbone apps that run in Node and the browser and consume our external API."
+This move from monolith to modular systems continues to influence day-to-day work on the Artsy Engineering team.
+
+We moved our API from Rails to [Grape][], partially motivated by a desire to build an iOS application that would
+consume this API. We faced a lot of problems with the split, including SEO problems, severe page load times,
+maintaining duplicated backend and frontend UI templates, slow test suites, and poor developer productivity. We
+took the project of building our mobile web frontend, m.artsy.net (still known as "martsy" internally) as an
+opportunity to address these problems.
+
+We built our new site with [Node.js][node] since it easily allowed server-side rendering. We split out areas of
```

Comment:
Let's clarify the `server-side rendering` bit. Maybe "since it allowed us to share and consolidate our server/client rendering code"